src/main/java/com/aws/greengrass/lifecyclemanager/Lifecycle.java

```diff
@@ -433,6 +436,16 @@ private void handleCurrentStateNew(Optional<State> desiredState) throws Interrup
 
         long currentStateGeneration = stateGeneration.incrementAndGet();
         replaceBackingTask(() -> {
+            // Wait until we're allowed to transition into INSTALLED
+            while (!stateTransitionAllowerService.isStateTransitionAllowed(greengrassService, State.NEW,
+                    State.INSTALLED)) {
+                try {
+                    Thread.sleep(STATE_TRANSITION_ALLOWED_POLLING_TIME_MILLIS);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    return;
+                }
+            }
```

Comment:
Possible to throw this in a method that takes the `from` -> `to` args? Looks the same as the one below.